### ISSUE TITLE
[GitHub Actions] Blocking a Github PR until UI Tests and UI Reviews are passing in a monorepo

### ISSUE THREAD

**Issue by VickyKoblinski** [chromaui/chromatic-cli#471]
I'm having a lot of trouble figuring this one out. Between the GitHub Action docs and the Chromatic docs, I'm a bit at a loss. I don't know where else to go, hopefully this is the right place.

We started with the suggested github action template recommended for monorepos:

```yml
# Workflow name
name: "Chromatic"

# Event for the workflow
on: push

# List of jobs
jobs:
  chromatic-deployment:
    # Operating System
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: frontend
    # Job steps
    steps:
      - uses: actions/checkout@v1
      - name: Install dependencies
        run: yarn
        working-directory: frontend
        # Adds Chromatic as a step in the workflow
      - name: Publish to Chromatic
        uses: chromaui/action@v1
        # Chromatic GitHub Action options
        with:
          workingDir: frontend
          token: ${{ secrets.GITHUB_TOKEN }}
          # Chromatic projectToken, refer to the manage page to obtain it.
          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
          onlyChanged: true
```

Which does work, however we have a lot more PRs that happen in `backend/` and a lot of merge commits and we're pushing to Chromatic a lot more often than we need. We have also required `UI Tests` and `UI Reviews` to pass, otherwise our developers were not bothering to look there.

We've thought about trying

```yml
on:
  # Chromatic needs the commit sha of master to use as image base for comparisons
  push:
    branches:
      - master
  # Only run if the PR contains changes to frontend/**
  pull_request:
    types: [opened, synchronize]
    paths:
      - "frontend/**"
```

or using a script like

```yml
    step:
      - name: Check if service has changed
        uses: actions/github-script@0.9.0
        id: service-changed
        with:
          result-encoding: string
          script: |
            const result = await github.pulls.listFiles({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.number,
              per_page: 100
            })
            const serviceChanged = result.data.filter(f => f.filename.startsWith("frontend/")).length > 0
            console.log(serviceChanged)
            return serviceChanged
```

to make sure the action always runs, but then to only run Chromatic if `serviceChanged` is true. But even if we can figure out that part, we run into a problem where now story UI Reviews is required, but it never runs. 

I also tried making a noop action, as suggested by the [GitHub Action docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks) but when making this action

```yml
name: "UI Review"

on:
  # Run if no changes to the frontend have been made
  pull_request:
    paths-ignore:
      - "frontend/**"

jobs:
  name: "UI Review"
  build:
    runs-on: ubuntu-latest
    steps:
      - run: 'echo "Chromatic not required" '
```

I run into the problem where this action's name really is `UI Review / build (pull_request)` which doesn't match `UI Review` so the PR is blocked until frontend changes are made.

Another thing we've tried but ran into problems was if if we do `on: push: paths: "frontend/**"`, we could have a PR that pushed code to the frontend and backend, and Chromatic would run (good), and then another push is done to the PR and the Chromatic action is skipped (bad). Then we have a Chromatic UI diff sitting in Chromatic but nothing in the PR shows that it exists. It doesn't get looked at and then the issue gets merged.

To sum this up, I have a two-pronged problem:

1. I am making Chromatic build way, way, way more often than we need. 
2. I need to Chromatic reviews to block a PR, otherwise they will not happen.

**Comment by tmeasday** [comment]
Hi @VickyKoblinski -- have you seen the `skip` option to Chromatic? I think maybe that can do exactly what you need here?

**Comment by VickyKoblinski** [comment]
You rock!! That helped a lot! The docs for `skip` are hard to track down if you don't know that the CLI and the action are tightly coupled. There's no documentation on the action for this.

This is what I needed to do:

```yml
# Workflow name
name: "Chromatic"

# Event for the workflow
on:
  # Chromatic needs the commit sha of master to use as image base for comparisons
  push:
    branches:
      - master
  # Only run if the PR contains changes to frontend/**
  pull_request:
    types: [opened, synchronize]

# List of jobs
jobs:
  chromatic-deployment:
    # Operating System
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: frontend
    # Job steps
    steps:
      - name: Check if service has changed
        uses: actions/github-script@0.9.0
        id: service-changed
        with:
          result-encoding: string
          script: |
            const result = await github.pulls.listFiles({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.number,
              per_page: 100
            })
            const serviceChanged = result.data.filter(f => f.filename.startsWith("frontend/")).length > 0
            console.log(serviceChanged)
            return serviceChanged

      - uses: actions/checkout@v1
      - name: Install dependencies
        run: yarn
        working-directory: frontend
        # Adds Chromatic as a step in the workflow
      - name: Publish to Chromatic
        uses: chromaui/action@v1
        # Chromatic GitHub Action options
        with:
          workingDir: frontend
          token: ${{ secrets.GITHUB_TOKEN }}
          # Chromatic projectToken, refer to the manage page to obtain it.
          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
          onlyChanged: true
          skip: ${{ steps.service-changed.outputs.result == 'false' }}
```

Here's some doc issues:
https://www.chromatic.com/docs/cli for `--skip` it doesn't mention a boolean. I was confused reading this, when I saw "Globs are supported" I thought it meant that branch names are supported OR pathnames via globs. I know it says "Only for branch". This was probably me projecting and really wishing it took in a path.
https://www.chromatic.com/docs/github-actions No mention of `skip` here, this is where I looked first. The docs for options here are definitely clearer here in my opinion than on the cli page.
https://github.com/chromaui/action No mention of `skip` here either, this was the second place I looked.

Overall, this works, thank you very much! I'm seeing https://github.com/chromaui/chromatic-cli/issues/351 this problem too which maybe the fix hasn't hit production yet?

Thank you though this solved my problem and got me exactly what I wanted!!

My one request would be it'd be super awesome if `skip` could support path globs. :)

**Comment by ghengeveld** [comment]
Hey @VickyKoblinski, what you're looking for is `--only-changed` in conjunction with [`--ignore-changed`,](https://github.com/chromaui/chromatic-cli/pull/461) the latter being a new feature that is going to be released soon. With that flag, you will be able to ignore certain paths, or (using a glob) everything but a certain path:
```
ignoreChanged: "!(frontend/**)"
```
This feature is going to take some QA but I expect it to land later this month.

**Comment by tmeasday** [comment]
@VickyKoblinski thanks for all the great feedback! I've created [PRs](https://github.com/chromaui/chromatic-cli/pull/473) to update the docs, and I opened this feature requested based on your request: https://github.com/chromaui/chromatic-cli/issues/472
